pinot-core/src/main/java/org/apache/pinot/core/query/optimizer/filter/NumericalFilterOptimizer.java

```diff
@@ -448,21 +449,31 @@ private static void rewriteRangeOperator(Expression range, FilterKind kind, int
     }
   }
 
-  /** @return true if expression is a column of numeric type */
-  private static boolean isNumericColumn(Expression expression, Schema schema) {
-    if (expression.getType() != ExpressionType.IDENTIFIER) {
-      // Expression can not be a column.
-      return false;
-    }
-
-    String column = expression.getIdentifier().getName();
-    FieldSpec fieldSpec = schema.getFieldSpecFor(column);
-    if (fieldSpec == null || !fieldSpec.isSingleValueField()) {
-      // Expression can not be a column name.
-      return false;
+  /** @return field data type extracted from the expression. null if we can't determine the type. */
+  @Nullable
+  private static FieldSpec.DataType getDataType(Expression expression, Schema schema) {
+    if (expression.getType() == ExpressionType.IDENTIFIER) {
+      String column = expression.getIdentifier().getName();
+      FieldSpec fieldSpec = schema.getFieldSpecFor(column);
+      if (fieldSpec != null && fieldSpec.isSingleValueField()) {
+        return fieldSpec.getDataType();
```

Comment:
if `fieldSpec` is null, then probably it implies that column  / identifier does not exist in the schema and we should throw error ?